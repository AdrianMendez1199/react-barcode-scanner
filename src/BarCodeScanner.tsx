import React, { useEffect } from 'react';
import { BrowserBarcodeReader, Result } from '@zxing/library';

const BarCodeScanner = (
  { onUpdate }: { onUpdate(err: unknown, result: Result | undefined): void  }
): JSX.Element => {
  const codeReader = new BrowserBarcodeReader();

  useEffect(() => {

    codeReader.decodeOnceFromVideoDevice(undefined, 'video')
      .then((result) => {
        onUpdate(null, result)
      }).catch((e) => {
        onUpdate(e, undefined)
      })

    return (): void => {
      codeReader.reset()
    }

  }, [codeReader])

  return (
    <div id="scanner-container">
      <video id="video" className="dbrScanner-video" playsInline />
    </div>
  )
};

export default BarCodeScanner;