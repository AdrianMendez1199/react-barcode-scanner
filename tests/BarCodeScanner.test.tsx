import React from 'react';
import { mount, shallow } from 'enzyme';
import BarCodeScanner from '../src/index';

describe('<BarCodeScanner />', () => {
    const callback = jest.fn();
    const wrapper = mount(<BarCodeScanner onUpdate={callback} />);

    test('should render correctly', () => {
        const wrapper = shallow(<BarCodeScanner onUpdate={callback} />);
        expect(wrapper).toMatchSnapshot();
    });


    test('should match the properties', () => {
        const props = wrapper.find('video').props();
        expect(props.id).toBe('video');
        expect(props.className).toBe('dbrScanner-video');
        expect(props.playsInline).toBe(true);
        expect(callback).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });
})