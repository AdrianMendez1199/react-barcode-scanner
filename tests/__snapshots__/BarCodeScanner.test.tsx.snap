// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarCodeScanner /> should match the properties 1`] = `
<BarCodeScanner
  onUpdate={
    [MockFunction] {
      "calls": Array [
        Array [
          [TypeError: Cannot read property 'getUserMedia' of undefined],
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    id="scanner-container"
  >
    <video
      className="dbrScanner-video"
      id="video"
      playsInline={true}
    />
  </div>
</BarCodeScanner>
`;

exports[`<BarCodeScanner /> should render correctly 1`] = `
<div
  id="scanner-container"
>
  <video
    className="dbrScanner-video"
    id="video"
    playsInline={true}
  />
</div>
`;
